pajbot/apiwrappers/seventv.py

```diff
@@ -2,72 +2,54 @@
 from pajbot.apiwrappers.response_cache import ListSerializer
 from pajbot.models.emote import Emote
 
+from requests import HTTPError
+
+from typing import Tuple
+
 
 class SevenTVAPI(BaseAPI):
     def __init__(self, redis):
-        super().__init__(base_url="https://api.7tv.app/v2/", redis=redis)
+        super().__init__(base_url="https://7tv.io/v3/", redis=redis)
 
     @staticmethod
     def parse_emotes(api_response_data):
-        def get_url(emote_id, size):
-            return f"https://cdn.7tv.app/emote/{emote_id}/{size}x"
+        def get_emote_urls(host) -> Tuple[dict[str, str], int, int]:
+            urls = {}
+            base_url = "https:" + host["url"]
+            emote_size = 1
+            max_width = 0
+            max_height = 0
+            for file in host["files"]:
+                if file["format"] == "WEBP":
+                    name = file["name"]
+                    urls[str(emote_size)] = f"{base_url}/{name}"
+                    if file["width"] > max_width:
+                        max_width = file["width"]
+                    if file["height"] > max_height:
+                        max_height = file["height"]
+                    emote_size += 1
+            return (urls, max_width, max_height)
```

Comment:
I think it would be a good idea to raise an exception here (instead of using bogus placeholder data) if none of the files were in the webp format